patches/minecraft/net/minecraft/world/WorldProvider.java.patch

```diff
@@ -103,7 +103,7 @@
 +    @Nullable
 +    public String getSaveFolder()
 +    {
-+        return (dimensionId.toString().equals("minecraft:overworld") ? null : "DIM" + dimensionId.toString());
++        return (dimensionId.toString().equals("minecraft:overworld") ? null : "DIM" + net.minecraftforge.fml.common.registry.ForgeRegistries.DIMENSIONS.getValue(dimensionId).getDimIntID());
```

Comment:
You can just keep the dimension instance in worldprovider instead of looking up again from the registry